nci/lib/watcher.rb

```diff
@@ -260,10 +260,23 @@ def run
         #end
       #end
 
-      bump_version
-
-      cmd.run('git --no-pager diff')
-      cmd.run("git commit -a -vv -m 'New release'")
+      # check if future_is_early and make sure correct series is applied and
+      # bump both release_* branches if required
+
+      if NCI.future_is_early!
+        bump_version(current_series)
+        cmd.run('git --no-pager diff')
+        cmd.run("git commit -a -vv -m 'New release'")
+      else
```

Comment:
I fee like there is a logic bug.

- If future_is_early only bump current
- If future_is_not early bump future and current
- But what if there is no future series at all?

Perhaps more importantly, why would it make a difference if future is early? Shouldn't it be

- If future_series is set then bump future and current
- Else bump only current